client/src/lib/api.js

```diff
@@ -48,6 +48,25 @@ api.getRelatedProducts = (product_id) => {
   return get(`products/${product_id}/related`);
 }
 
+  // AGGREGATE DETAILS, STYLES, AND META DATA FOR PRODUCT
+  api.collectProductInfo = (id) => {
+    let product = {};
+
+    return api.getProductId(id)
+    .then((data) => {
+      product.details = data;
+      return api.getProductStyles(id);
+    })
+    .then((data) => {
+      product.styles = data;
+      return api.getReviewsMetadata({product_id: id});
+    })
+    .then((data) => {
+      product.reviewsMeta = data;
+      return product;
+    })
+    .catch(err => console.log(err));
+  }
```

Comment:
nice solution! this looks great. I wonder if using await would clean it up even more. E.g.:
`product.details = await api.getProductId(id)`
`product.styles = await api.getProductStyles(id)`
`product.reviewsMeta = await api.getReviewsMetadata({product_id: id})`

what do you think? 13 lines -> 3 lines seems pretty sweet :) I might do the same with some of my promises!